### ISSUE TITLE
Weird behavior of 3.1.9 on node 8.x

### ISSUE THREAD

**Issue by bkimminich** [mapbox/node-sqlite3#865]
Hi! When my Travis-CI build tries to do `npm install` for my application I everything works fine at first: https://travis-ci.org/bkimminich/juice-shop/jobs/264803693#L271

```
> sqlite3@3.1.9 install /home/travis/build/bkimminich/juice-shop/node_modules/sqlite3
> node-pre-gyp install --fallback-to-build
[sqlite3] Success: "/home/travis/build/bkimminich/juice-shop/node_modules/sqlite3/lib/binding/node-v57-linux-x64/node_sqlite3.node" is installed via remote
```

But then in the same build during the test execution `npm install` is called again and it now fails:
https://travis-ci.org/bkimminich/juice-shop/jobs/264803693#L5700

```
npm ERR! path /home/travis/build/bkimminich/juice-shop/node_modules/.staging/npmlog-a0d66ba0
npm ERR! code ENOENT
npm ERR! errno -2
npm ERR! syscall rename
npm ERR! enoent ENOENT: no such file or directory, rename '/home/travis/build/bkimminich/juice-shop/node_modules/.staging/npmlog-a0d66ba0' -> '/home/travis/build/bkimminich/juice-shop/node_modules/sqlite3/node_modules/node-pre-gyp/node_modules/npmlog'
npm ERR! enoent This is related to npm not being able to find a file.
npm ERR! enoent 
npm ERR! A complete log of this run can be found in:
npm ERR!     /home/travis/.npm/_logs/2017-08-15T17_06_53_744Z-debug.log
npm ERR! code ELIFECYCLE
npm ERR! errno 254
npm ERR! juice-shop@5.0.0-SNAPSHOT preupdate-webdriver: `npm install`
npm ERR! Exit status 254
npm ERR! 
npm ERR! Failed at the juice-shop@5.0.0-SNAPSHOT preupdate-webdriver script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
0m
npm ERR! A complete log of this run can be found in:
npm ERR!     /home/travis/.npm/_logs/2017-08-15T17_06_53_772Z-debug.log
npm ERR! code ELIFECYCLE
npm ERR! errno 254
npm ERR! juice-shop@5.0.0-SNAPSHOT preprotractor: `npm run update-webdriver`
npm ERR! Exit status 254
npm ERR! 
npm ERR! Failed at the juice-shop@5.0.0-SNAPSHOT preprotractor script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
npm ERR! A complete log of this run can be found in:
npm ERR!     /home/travis/.npm/_logs/2017-08-15T17_06_53_785Z-debug.log
The command "NODE_ENV=ctf npm run protractor" exited with 254.
Done. Your build exited with 1.
```

This only happens during the node.js v8 build, not on v6. Do you have any idea why this might be happening? All help is greatly appreciated!

**Comment by JKathmann** [comment]
I have the same issue:
```
➜  ~ mkdir sqlite3test
➜  ~ cd sqlite3test
➜  sqlite3test node --version
v8.3.0
➜  sqlite3test npm --version
5.3.0
➜  sqlite3test npm init
➜  sqlite3test npm install sqlite3
... (after a while)
+ sqlite3@3.1.9
added 134 packages
➜  sqlite3test npm install
npm WARN sqlite3test@1.0.0 No description
npm WARN sqlite3test@1.0.0 No repository field.

npm ERR! path /root/sqlite3test/node_modules/.staging/npmlog-7f4dfe8f
npm ERR! code ENOENT
npm ERR! errno -2
npm ERR! syscall rename
npm ERR! enoent ENOENT: no such file or directory, rename '/root/sqlite3test/node_modules/.staging/npmlog-7f4dfe8f' -> '/root/sqlite3test/node_modules/sqlite3/node_modules/node-pre-gyp/node_modules/npmlog'
npm ERR! enoent This is related to npm not being able to find a file.
npm ERR! enoent

npm ERR! A complete log of this run can be found in:
npm ERR!     /root/.npm/_logs/2017-08-16T11_35_42_031Z-debug.log
```

**Comment by bkimminich** [comment]
I pinned to `"sqlite3": "3.1.8"` and that version works on node 8.x. I consider this a workaround only, but I'll relabel the ticket to clarify the problem is in 3.1.9 exclusively.

**Comment by JKathmann** [comment]
confirmed:
```
➜  npm install sqlite3@3.1.8
...
+ sqlite3@3.1.8
added 143 packages
➜  npm install
npm WARN sqlite3test2@1.0.0 No description
npm WARN sqlite3test2@1.0.0 No repository field.

up to date
➜
```

**Comment by springmeyer** [comment]
This is a dupe of https://github.com/mapbox/node-sqlite3/issues/866 I think, so going to close and track at https://github.com/mapbox/node-sqlite3/issues/866#issuecomment-327947761

**Comment by sunnykgupta** [comment]
https://npm.community/t/npm-install-for-package-with-local-dependency-fails/754/7

Solved in npm@next (v6.9.0)
